### PR TITLE
Checks refactor

### PR DESCRIPTION
**Pull request by gusinacio**
Breaking Change: timestamp check is now the responsibility of the lib user.

### CODE REVIEW

File: tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    executor_mock: ExecutorFixture,
+) {
+    let ExecutorFixture {
+        checks,
+        executor,
+        escrow_storage,
+        query_appraisals,
+        ..
+    } = executor_mock;
+    // give receipt 5 second variance for min start time
```
Comment (gusinacio) [inline tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs:234]: remove comment

File: tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    let signed_receipt = EIP712SignedMessage::new(
+        &domain_separator,
+        Receipt::new(allocation_ids[0], query_value).unwrap(),
+        &keys.0,
+    )
+    .unwrap();
+
+    let query_id = 1;
+
+    // prepare adapters and storage to correctly validate receipt
```
Comment (gusinacio) [inline tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs:247]: remove comment

File: tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+    executor_mock: ExecutorFixture,
+) {
+    let ExecutorFixture {
+        checks,
+        executor,
+        escrow_storage,
+        query_appraisals,
+        ..
+    } = executor_mock;
+    // give receipt 5 second variance for min start time
```
Comment (gusinacio) [inline tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs:300]: remove comment

File: tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+    let signed_receipt = EIP712SignedMessage::new(
+        &domain_separator,
+        Receipt::new(allocation_ids[0], query_value).unwrap(),
+        &keys.0,
+    )
+    .unwrap();
+
+    let query_id = 1;
+
+    // prepare adapters and storage to correctly validate receipt
```
Comment (gusinacio) [inline tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs:313]: remove comment

File: tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
```
Comment (gusinacio) [inline tap_core/src/tap_receipt/received_receipt/received_receipt_unit_test.rs:185]: remove comment

File: tap_core/src/checks/mod.rs [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::tap_receipt::{Checking, ReceiptError, ReceiptResult, ReceiptWithState};
+use serde::{Deserialize, Serialize};
+use std::sync::{Arc, RwLock};
+
+pub type ReceiptCheck = Arc<dyn Check>;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum CheckingChecks {
```
Comment (aasseman) [inline tap_core/src/checks/mod.rs:11]: `CheckingChecks` is a pretty weird name. Would something more direct such as `CheckWithState` make sense?

**Review by aasseman — CHANGES_REQUESTED** [review]
LGTM, apart from one Struct name that I find confusing.

Otherwise, I don't remember if you wanted to remove serde from ReceivedReceipts in this PR. Feel free to throw it in here if you'd like :shrug:

File: tap_core/src/checks/mod.rs (hunk 6 above)
Comment (gusinacio) [reply tap_core/src/checks/mod.rs:11]: I'll remove in the next PR, so it doesn't make sense to rename now.

**Review by aasseman — APPROVED** [review]
Concerns will be tackled in next PRs

**Pull request merged**